### PR TITLE
refactor(SettingsAddressBook): code

### LINKED ISSUE #389: settings-address-book

### PR DESCRIPTION
**Pull request by samandarkuchkarov**
Fixes #389 

## Proposed Changes

-Change the imports to absolute:
-Move text to i18n
-Change the colors from the constants to similar ones in Color
-Transferring the screen logic to components
-Replace navigation hooks with typed navigation hooks in folder src/hooks

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/components/settings-address-book.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        style={styles.input}
+        placeholder={getText(I18N.settingsAddressBookPlaceholder)}
+        value={search}
+        onChangeText={setSearch}
+        multiline
+        autoFocus
+        rightAction={
+          search === '' ? (
+            <IconButton onPress={onPressQR}>
+              <QRScanner color={LIGHT_GRAPHIC_GREEN_1} style={styles.icon} />
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:59]: Replace this with an Icon component with the appropriate size
```jsx
<Icon i24 name="qr_scanner" color={Color.graphicGreen1} />
```

File: src/components/settings-address-book.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            </IconButton>
+          ) : (
+            <IconButton onPress={onPressClear}>
+              <Icon i24 name="close_circle" color={LIGHT_GRAPHIC_BASE_2} />
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:63]: Use `Color` enum in icons color

File: src/components/settings-address-book.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          )
+        }
+      />
+
+      {canAdd && (
+        <IconButton onPress={onPressAdd} style={styles.addButton}>
+          <Box style={styles.badge}>
+            <Icon i24 name="plus_mid" color={LIGHT_GRAPHIC_BASE_2} />
+          </Box>
+          <Text color={LIGHT_TEXT_BASE_1} i18n={I18N.settingsAddressBookAdd} />
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:74]: Use `Color` enum in text color prop

File: src/components/settings-address-book.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    </>
+  );
+};
+
+const styles = createTheme({
+  input: {
+    marginBottom: 12,
+    marginHorizontal: 20,
+  },
+  icon: {width: 20, height: 20},
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:93]: Remove this and use `i24` prop in Icons

File: src/components/settings-address-book.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  onPressAdd,
+  onPressClear,
+  onPressQR,
+  canAdd,
+  search,
+  setSearch,
+}: SettingsAddressBookProps) => {
+  return (
+    <>
+      <TextField
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:48]: Fix colors in `TextField`

File: src/screens/settings-address-book.tsx [hunk 6]
```diff
@@ -2,35 +2,13 @@ import React, {useCallback, useEffect, useState} from 'react';
 
 import {CompositeScreenProps} from '@react-navigation/native';
 import {utils} from 'ethers';
-import {StyleSheet} from 'react-native';
 
+import {SettingsAddressBook} from '@app/components/settings-address-book';
+import {hideModal, showModal} from '@app/helpers/modal';
 import {useApp, useContacts} from '@app/hooks';
+import {useTypedNavigation} from '@app/hooks';
```
Comment (gHashTag) [inline src/screens/settings-address-book.tsx:9]: use one import for hooks

**Review by gHashTag — CHANGES_REQUESTED** [review]
Pls fix comments

File: src/components/settings-address-book.tsx (hunk 5 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx:48]: do you mean colors in icon and text ?

File: src/screens/settings-address-book.tsx (hunk 6 above)
Comment (samandarkuchkarov) [reply src/screens/settings-address-book.tsx]: done

File: src/components/settings-address-book.tsx (hunk 5 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx:44]: done

File: src/components/settings-address-book.tsx (hunk 4 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx:92]: done

File: src/components/settings-address-book.tsx (hunk 3 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx]: done

File: src/components/settings-address-book.tsx (hunk 2 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx]: done

File: src/components/settings-address-book.tsx (hunk 1 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx]: done

**Comment by samandarkuchkarov** [comment]
> Pls fix comments

done

File: src/components/settings-address-book.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        style={styles.input}
+        placeholder={getText(I18N.settingsAddressBookPlaceholder)}
+        value={search}
+        onChangeText={setSearch}
+        multiline
+        autoFocus
+        rightAction={
+          search === '' ? (
+            <IconButton onPress={onPressQR}>
+              <Icon name="qr_scanner" color={Color.graphicGreen1} />
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:48]: use `i24` prop for size definitions

File: src/components/settings-address-book.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+            </IconButton>
+          ) : (
+            <IconButton onPress={onPressClear}>
+              <Icon name="close_circle" color={getColor(Color.graphicBase2)} />
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:52]:
```jsx
<Icon i24 name="close_circle" color={Color.graphicBase2} />
```

File: src/components/settings-address-book.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        }
+      />
+
+      {canAdd && (
+        <IconButton onPress={onPressAdd} style={styles.addButton}>
+          <Box style={styles.badge}>
+            <Icon name="plus_mid" color={getColor(Color.graphicBase2)} />
+          </Box>
+          <Text
+            color={getColor(Color.textBase2)}
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:64]:
```jsx
color={Color.textBase2}
```

File: src/components/settings-address-book.tsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+              <Icon name="close_circle" color={getColor(Color.graphicBase2)} />
+            </IconButton>
+          )
+        }
+      />
+
+      {canAdd && (
+        <IconButton onPress={onPressAdd} style={styles.addButton}>
+          <Box style={styles.badge}>
+            <Icon name="plus_mid" color={getColor(Color.graphicBase2)} />
```
Comment (gHashTag) [inline src/components/settings-address-book.tsx:61]: Use i24 prop for size definitions
`color` prop without `getColor` function call

File: src/screens/settings-address-book.tsx [hunk 11]
```diff
@@ -84,60 +61,15 @@ export const SettingsAddressBookScreen =
 
     return (
       <>
```
Comment (gHashTag) [inline src/screens/settings-address-book.tsx:63]: Unnecessary `React.Fragment`, remove pls

File: src/components/settings-address-book.tsx (hunk 5 above)
Comment (gHashTag) [reply src/components/settings-address-book.tsx:37]: I mean, you need to tidy up the `TextField` code
In order for the color themes to work

**Review by gHashTag — CHANGES_REQUESTED** [review]
Please correct the comments, you did a good job🙌

File: src/screens/settings-address-book.tsx (hunk 11 above)
Comment (samandarkuchkarov) [reply src/screens/settings-address-book.tsx]: done

File: src/components/settings-address-book.tsx (hunk 8 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx]: done

File: src/components/ui/text-field.tsx [hunk 12]
```diff
@@ -240,7 +241,7 @@ const styles = StyleSheet.create({
   },
   placeholder: {
     position: 'absolute',
-    color: PLACEHOLDER_GRAY,
```
Comment (gHashTag) [inline src/components/ui/text-field.tsx:243]: If there are no colors in the Color enum, then the constants should be left

Comment (samandarkuchkarov) [reply src/components/ui/text-field.tsx]: done

File: src/components/settings-address-book.tsx (hunk 5 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx:37]: done

File: src/components/settings-address-book.tsx (hunk 7 above)
Comment (samandarkuchkarov) [reply src/components/settings-address-book.tsx]: done

**Comment by gHashTag** [comment]
Move ListFooterComponent definition out of the parent component “ListContact” and pass data as props. In `src/components/list-contact.tsx` 33 line

**Review by gHashTag — APPROVED** [review]

File: src/hocs/with-actions-contact-item.tsx [hunk 13]
```diff
@@ -68,24 +69,24 @@ export const withActionsContactItem = (
...
-              text: 'Delete',
-              style: 'destructive',
-              onPress: () => {
-                const contactAddress =
-                  typeof itemOrAddress === 'string'
-                    ? itemOrAddress
-                    : 'account' in itemOrAddress && itemOrAddress.account;
-                contactAddress && contacts.removeContact(contactAddress);
-              },
+        const AlertTitle = getText(I18N.settingsAddressBookAlertTitle);
```
Comment (vivalaakam) [inline src/hocs/with-actions-contact-item.tsx:72]: pls change name variables to camelCase

**Review by vivalaakam — CHANGES_REQUESTED** [review]
discard [src/components/settings-about.tsx](https://github.com/haqq-network/haqq-wallet/pull/532/files#diff-e4a6abb48aaf12caadd8880fe4816fa03b5e0fb68c5a00a16983a8b9743f5742) pls, cause you're working on setting address component

File: src/hocs/with-actions-contact-item.tsx (hunk 13 above)
Comment (samandarkuchkarov) [reply src/hocs/with-actions-contact-item.tsx]: done

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**